File: java/src/main/java/com/opl/aircraftperformancecalculator/models/RunwayConditions.java
package com.opl.aircraftperformancecalculator.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.Id;

/**
 * Class to represent current runway conditions.
 */
@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RunwayConditions {

    @Id
    private String airportID;

    private double temp;
    private double pressureAltitude;
    private double precipitation;
    private double headWind;
    private double runwayLength;
    private String runwayType; // can be concrete or grass, but represented as "CONC"/"ASPH" or "TURF"
    private double slope;
}

File: FrontEnd/src/app/models/runway-conditions.ts
export class RunwayConditions {
    public airportID: string
    public temp: number;
    public pressureAltitude: number;
    public precipitation: number;
    public headWind: number;
    public runwayLength: number;
    public runwayType: string;
    public slope: number;

    constructor(airportID: string,temp: number, pressureAltitude: number, precipitation: number, headWind: number, runwayLength: number, runwayType: string, slope: number) {
        this.airportID = airportID
        this.temp = temp;
        this.pressureAltitude = pressureAltitude;
        this.precipitation = precipitation;
        this.headWind = headWind;
        this.runwayLength =  runwayLength;
        this.runwayType = runwayType;
        this.slope = slope;
    }
}